src/workloads/execution/FilterWithComplexLogicalExpression.yml

```diff
@@ -426,6 +434,535 @@ Actors:
+  Type: CrudActor
+  Database: *Database
+  Threads: *Threads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [33]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: *Repeat
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {$expr: {$eq: ["$a.b", 0]}}
+
+- Name: MatchExpressionArrayStressTest
+  Type: CrudActor
+  Database: *Database
+  Threads: *Threads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [34]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: *Repeat
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {"a.b": 0}
+
+# Clean up the data from the previous phase.
+- Name: ClearCollectionArrayStressTestData
+  Type: CrudActor
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [35]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: QuiesceCollectionDropArrayStressTestData
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [36]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+
+# Insert documents with 2 levels of nested arrays where the leaves contain many fields. This test
+# targets workloads where the paths queried share a common prefix ("a.b.c", "a.b.d", and so on).
+- Name: InsertRepeatedPathTraversalData
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [37]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document: {a: [{^Repeat: {count: 3, fromGenerator: {b: [{c: 0, d: 0, e: 0, f: 0, g: 0, h: 0, i: 0, j: 0, k: 0, l: 0}]}}}]}
+
+- Name: QuiesceRepeatedPathTraversalData
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [38]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+
+- Name: AggregationExpressionRepeatedPathTraversal
+  Type: CrudActor
+  Database: *Database
+  Threads: *Threads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [39]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: *Repeat
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {$expr: {$or: [{$eq: ["$a.b.c", 1]},
+                                   {$eq: ["$a.b.d", 1]},
+                                   {$eq: ["$a.b.e", 1]},
+                                   {$eq: ["$a.b.f", 1]},
+                                   {$eq: ["$a.b.g", 1]}]}}
```

Comment:
Add a test case like `{$ne: ["$a.b.c", "$a.b.d"]}`?

Reply:
My idea is to have path field in both side of binary op, instead of a constant in one side.

Reply:
Ah, good idea. Done (only for the agg versions)